Fixes infinite loop in TailedFile with long lines.

foxdan opened the pull request:
If a long line is encountered where the trailing newline is not in the buffer
and the buffer is not considered used enough to be cleaned an infinite loop
occurs.

Fix this by ignoring such lines and looping through until the next line is
found.

derpston commented:
I'm not thrilled with the idea of skipping content, but this behaviour is preferable to a complete failure to process any more content at all.

This was time consuming to review without tests and I still have a relatively low level of confidence. What sort of testing have you given this, @foxdan?

foxdan commented:
Fair point, it was tested by running over a file without `skip_to_end` that was previously causing lockups.

Added a unittest for this and in fact it was returning the end of the longline, now fixed.

derpston commented:
OK, thanks.
